src/utils/storage.ts

```diff
@@ -0,0 +1,83 @@
+import { ImporterState } from '../importer/types';
+import { SheetDefinition } from '../sheet/types';
+
+const DB_NAME = 'HelloCSV';
+const DB_VERSION = 1;
+const STORE_NAME = 'state';
+
+export async function getIndexedDBState(
+  sheetDefinitions: SheetDefinition[]
+): Promise<ImporterState> {
+  return new Promise((resolve, reject) => {
+    const key = getStateKey(sheetDefinitions);
+    const request = indexedDB.open(DB_NAME, DB_VERSION);
+
+    request.onerror = () => reject(request.error);
+    request.onsuccess = () => {
+      const db = request.result;
+      const transaction = db.transaction(STORE_NAME, 'readonly');
+      const store = transaction.objectStore(STORE_NAME);
+      const getRequest = store.get(key);
+
+      getRequest.onerror = () => reject(getRequest.error);
+      getRequest.onsuccess = () => {
+        try {
+          const result = getRequest.result;
+          result.sheetDefinitions = sheetDefinitions;
```

Comment:
I am wondering if this won't cause some problems - like what if e.g. integrator changes some definition (e.g. id of some sheet)? We store currentSheetId in the state and if integrator changes the id in the definition, then won't that cause the app to crash, since sheet with currentSheetId doesn't exist?

Reply:
if the integrator changes a sheetDefinition, we would create a new row in the indexDB database